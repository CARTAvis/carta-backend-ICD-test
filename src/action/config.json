{
    "serverURL": "wss://carta.asiaa.sinica.edu.tw/socketdev",
    "serverURL0": "ws://127.0.0.1:3002",
    "serverURL1": "ws://cartaviewer:5000",
    "path": {
        "root": ".",
        "base": "$BASE",
        "directory": ".",
        "QA": "set_QA",
        "region": "set_QA/set_QA_regionTest",
        "performance": "set_QA_performance"
    },
    "image": {
        "singleChannel": "cube_A/cube_A_01600_z00001.fits",
        "cube": "cube_A/cube_A_00800_z00100.fits"
    },
    "timeout": {
        "connection": 1000,
        "listFile": 1000,
        "openFile": 1000,
        "readFile": 6100,
        "readLargeImage": 60000,
        "changeChannel": 3000,
        "playImages": 8000,
        "cubeHistogram": 500000,
        "cancel": 5000,
        "concurrent": 4000,
        "messageEvent": 500,
        "mouseEvent": 400,
        "region": 500,
        "import": 500,
        "export": 500,
        "resume": 10000,
        "sleep": 3000,
        "playAnimator": 140000,
        "playLargeAnimator": 600000,
        "contour": 4000,
        "zoom": 12200
    },
    "repeat": {
        "concurrent": 10,
        "cursor": 20,
        "image": 10,
        "animation": 10,
        "resumeSession": 10,
        "contour": 10,
        "zoom": 5
    },
    "log": {
        "error": false,
        "warning": true,
        "event": false
    }
}