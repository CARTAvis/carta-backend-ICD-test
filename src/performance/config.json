{    
    "serverURL": "ws://127.0.0.1",
    "port": 7001,
    "timeout": {
        "connection": 2000,
        "openFile": 20000,
        "readFile": 40000
    },
    "wait": {
        "exec": 500,
        "ps": 1000,
        "reconnect": 30,
        "event": 35,
        "cursor": 35
    },
    "repeat": {
        "cursor": 20,
        "event": 10
    },
    "path": {
        "backend": "/Users/zarda/GitHub/carta-backend-nrao/build",
        "performanceTestResultDir": "/Users/zarda/GitHub/carta-backend-ICD-test/performance_result",
        "base": "$HOME/CARTA/Images",
        "performance": "set_QA_performance"
    },
    "log" : false
}