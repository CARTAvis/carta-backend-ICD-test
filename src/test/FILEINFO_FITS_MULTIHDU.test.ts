import { CARTA } from "carta-protobuf";
import { Client } from "./CLIENT";
import config from "./config.json";

let testServerUrl = config.serverURL;
let testSubdirectory = config.path.QA;
let connectTimeout = config.timeout.connection;
let listFileTimeout = config.timeout.listFile;
let openFileTimeout = 100;//config.timeout.openFile;

interface AssertItem {
    register: CARTA.IRegisterViewer;
    filelist: CARTA.IFileListRequest;
    fileInfoRequest: CARTA.IFileInfoRequest[];
    fileInfoResponse: CARTA.IFileInfoResponse[];
};

let assertItem: AssertItem = {
    register: {
        sessionId: 0,
        clientFeatureFlags: 5,
    },
    filelist: { directory: testSubdirectory },
    fileInfoRequest: [
        {
            directory: testSubdirectory,
            file: "spire500_ext.fits",
            hdu: "1 ExtName: image",
        },
    ],
    fileInfoResponse: [
        {
            success: true,
            message: "",
            fileInfo: {
                name: "spire500_ext.fits",
                type: CARTA.FileType.FITS,
                size: 17591040,
                HDUList: ["1 ExtName: image   ", "6 ExtName: error   ", "7 ExtName: coverage"],
            },
            fileInfoExtended: {
                dimensions: 2,
                width: 830,
                height: 870,
                depth: 1,
                stokes: 1,
                stokesVals: [],
                computedEntries: [
                    { name: "Name", value: "spire500_ext.fits" },
                    { name: "Shape", value: "[830, 870]" },
                    { name: "Coordinate type", value: "Right Ascension, Declination" },
                    { name: "Projection", value: "TAN" },
                    { name: "Image reference pixels", value: "[371, 421]" },
                    { name: "Image reference coords", value: "[07:09:13.1081, -010.36.38.5952]" },
                    { name: "Image ref coords (deg)", value: "[107.305 deg, -10.6107 deg]" },
                    { name: "Celestial frame", value: "FK5, J2000" },
                    { name: "Pixel unit", value: "MJy/sr" },
                    { name: "Pixel increment", value: "-14\", 14\"" },
                ],
                headerEntries: [
                    { name: "SIMPLE", value: "T / Standard FITS", entryType: 2, numericValue: 1 },
                    { name: "BITPIX", value: "-32 / Floating point (32 bit)", entryType: 2, numericValue: -32 },
                    { name: "NAXIS", value: "2", entryType: 2, numericValue: 2 },
                    { name: "NAXIS1", value: "830", entryType: 2, numericValue: 830 },
                    { name: "NAXIS2", value: "870", entryType: 2, numericValue: 870 },
                    {
                        name: "BSCALE",
                        value: "1.000000000000E+00 / PHYSICAL = PIXEL*BSCALE + BZERO",
                        entryType: 1,
                        numericValue: 1
                    },
                    {
                        name: "BZERO",
                        value: "0.000000000000E+00",
                        entryType: 1,
                    },
                    { name: "BTYPE", value: "Intensity" },
                    { name: "OBJECT" },
                    { name: "BUNIT", value: "MJy/sr / Brightness (pixel) unit" },
                    {
                        name: "EQUINOX",
                        value: "2000",
                        entryType: 1,
                        numericValue: 2000
                    },

                    { name: "RADESYS", value: "FK5" },
                    {
                        name: "LONPOLE",
                        value: "1.800000000000E+02",
                        entryType: 1,
                        numericValue: 180
                    },
                    {
                        name: "LATPOLE",
                        value: "-1.061072089652E+01",
                        entryType: 1,
                        numericValue: -10.610720896516849
                    },
                    {
                        name: "PC1_1",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "PC2_1", value: "-0.000000000000E+00", entryType: 1 },
                    { name: "PC1_2", value: "0.000000000000E+00", entryType: 1 },
                    {
                        name: "PC2_2",
                        value: "1.000000000000E+00",
                        entryType: 1,
                        numericValue: 1
                    },
                    { name: "CTYPE1", value: "RA---TAN" },
                    {
                        name: "CRVAL1",
                        value: "1.073046172702E+02",
                        entryType: 1,
                        numericValue: 107.30461727023817
                    },
                    {
                        name: "CDELT1",
                        value: "-3.888888888889E-03",
                        entryType: 1,
                        numericValue: -0.003888888888889
                    },
                    { name: "CRPIX1", value: "371", entryType: 1, numericValue: 371 },
                    { name: "CUNIT1", value: "deg" },
                    { name: "CTYPE2", value: "DEC--TAN" },
                    {
                        name: "CRVAL2",
                        value: "-1.061072089652E+01",
                        entryType: 1,
                        numericValue: -10.610720896516849
                    },
                    {
                        name: "CDELT2",
                        value: "3.888888888889E-03",
                        entryType: 1,
                        numericValue: 0.003888888888889
                    },
                    { name: "CRPIX2", value: "421", entryType: 1, numericValue: 421 },
                    { name: "CUNIT2", value: "deg" },
                    { name: "LONGSTRN", value: "OGIP 1.0" },
                    { name: "EXTNAME", value: "image" },
                    { name: "CLASS___", value: "herschel.ia.dataset.ArrayDataset" },
                    { name: "INFO____", value: "Image" },
                    { name: "DATA____", value: "herschel.ia.numeric.Double2d" },
                    { name: "QTTY____", value: "MJy/sr" },
                    {
                        name: "META_0",
                        value: "2",
                        entryType: 2,
                        numericValue: 2
                    },
                    {
                        name: "META_1",
                        value: "830",
                        entryType: 2,
                        numericValue: 830
                    },
                    {
                        name: "META_2",
                        value: "870",
                        entryType: 2,
                        numericValue: 870
                    },
                    { name: "TIMESYS", value: "UTC / Time system for HDU" },
                ],
            },
        },
    ],
};

describe("FILEINFO test: Testing if info of an image file is correctly delivered by the backend", () => {

    let Connection: Client;
    beforeAll(async () => {
        Connection = new Client(testServerUrl);
        await Connection.open();
        await Connection.send(CARTA.RegisterViewer, assertItem.register);
        await Connection.receive(CARTA.RegisterViewerAck);
    }, connectTimeout);

    describe(`Go to "${testSubdirectory}" folder`, () => {
        beforeAll(async () => {
            await Connection.send(CARTA.FileListRequest, assertItem.filelist);
            await Connection.receive(CARTA.FileListResponse);
        }, listFileTimeout);

        describe(`query the info of file : ${assertItem.fileInfoRequest[0].file}`, () => {
            let FileInfoResponseTemp: CARTA.FileInfoResponse;
            test(`FILE_INFO_RESPONSE should arrive within ${openFileTimeout} ms".`, async () => {
                await Connection.send(CARTA.FileInfoRequest, assertItem.fileInfoRequest[0]);
                FileInfoResponseTemp = await Connection.receive(CARTA.FileInfoResponse);
            }, openFileTimeout);

            test("FILE_INFO_RESPONSE.success = true", () => {
                expect(FileInfoResponseTemp.success).toBe(true);
            });

            test(`FILE_INFO_RESPONSE.file_info.HDU_List = [${assertItem.fileInfoResponse[0].fileInfo.HDUList}]`, () => {
                assertItem.fileInfoResponse[0].fileInfo.HDUList.map((entry, index) => {
                    expect(FileInfoResponseTemp.fileInfo.HDUList).toContainEqual(entry);
                });
            });

            test(`FILE_INFO_RESPONSE.file_info.name = "${assertItem.fileInfoResponse[0].fileInfo.name}"`, () => {
                expect(FileInfoResponseTemp.fileInfo.name).toEqual(assertItem.fileInfoResponse[0].fileInfo.name);
            });

            test(`FILE_INFO_RESPONSE.file_info.size = ${assertItem.fileInfoResponse[0].fileInfo.size}`, () => {
                expect(FileInfoResponseTemp.fileInfo.size.toString()).toEqual(assertItem.fileInfoResponse[0].fileInfo.size.toString());
            });

            test(`FILE_INFO_RESPONSE.file_info.type = ${CARTA.FileType.FITS}`, () => {
                expect(FileInfoResponseTemp.fileInfo.type).toBe(assertItem.fileInfoResponse[0].fileInfo.type);
            });

            test(`FILE_INFO_RESPONSE.file_info_extended.dimensions = ${assertItem.fileInfoResponse[0].fileInfoExtended.dimensions}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.dimensions).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.dimensions);
            });

            test(`FILE_INFO_RESPONSE.file_info_extended.width = ${assertItem.fileInfoResponse[0].fileInfoExtended.width}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.width).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.width);
            });

            test(`FILE_INFO_RESPONSE.file_info_extended.height = ${assertItem.fileInfoResponse[0].fileInfoExtended.height}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.height).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.height);
            });

            if (assertItem.fileInfoResponse[0].fileInfoExtended.dimensions > 2) {
                test(`FILE_INFO_RESPONSE.file_info_extended.depth = ${assertItem.fileInfoResponse[0].fileInfoExtended.depth}`, () => {
                    expect(FileInfoResponseTemp.fileInfoExtended.depth).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.depth);
                });
            };

            if (assertItem.fileInfoResponse[0].fileInfoExtended.dimensions > 3) {
                test(`FILE_INFO_RESPONSE.file_info_extended.stokes = ${assertItem.fileInfoResponse[0].fileInfoExtended.stokes}`, () => {
                    expect(FileInfoResponseTemp.fileInfoExtended.stokes).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.stokes);
                });
            };

            test(`FILE_INFO_RESPONSE.file_info_extended.stokes_vals = [${assertItem.fileInfoResponse[0].fileInfoExtended.stokesVals}]`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.stokesVals).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.stokesVals);
            });

            test(`len(FILE_INFO_RESPONSE.file_info_extended.computed_entries)==${assertItem.fileInfoResponse[0].fileInfoExtended.computedEntries.length}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.computedEntries.length).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.computedEntries.length);
            });

            test(`assert FILE_INFO_RESPONSE.file_info_extended.computed_entries`, () => {
                assertItem.fileInfoResponse[0].fileInfoExtended.computedEntries.map((entry: CARTA.IHeaderEntry, index) => {
                    expect(FileInfoResponseTemp.fileInfoExtended.computedEntries).toContainEqual(entry);
                });
            });

            test(`len(file_info_extended.header_entries)==${assertItem.fileInfoResponse[0].fileInfoExtended.headerEntries.length}`, () => {
                expect(FileInfoResponseTemp.fileInfoExtended.headerEntries.length).toEqual(assertItem.fileInfoResponse[0].fileInfoExtended.headerEntries.length)
            });

            test(`assert FILE_INFO_RESPONSE.file_info_extended.header_entries`, () => {
                assertItem.fileInfoResponse[0].fileInfoExtended.headerEntries.map((entry: CARTA.IHeaderEntry, index) => {
                    expect(FileInfoResponseTemp.fileInfoExtended.headerEntries).toContainEqual(entry);
                });
            });
        });
    });
    afterAll(() => Connection.close());
});