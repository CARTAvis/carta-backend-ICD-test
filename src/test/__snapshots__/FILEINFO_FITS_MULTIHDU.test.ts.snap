// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FILEINFO_FITS_MULTIHDU: Testing if info of an FITS image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : spire500_ext.fits FileInfoExtended of '1': assert FILE_INFO_RESPONSE.file_info_extended.computed_entries 1`] = `
Array [
  Object {
    "name": "Name",
    "value": "spire500_ext.fits",
  },
  Object {
    "name": "HDU",
    "value": "1",
  },
  Object {
    "name": "Extension name",
    "value": "image",
  },
  Object {
    "name": "Shape",
    "value": "[830, 870]",
  },
  Object {
    "name": "Coordinate type",
    "value": "Right Ascension, Declination",
  },
  Object {
    "name": "Projection",
    "value": "TAN",
  },
  Object {
    "name": "Image reference pixels",
    "value": "[371, 421]",
  },
  Object {
    "name": "Image reference coords",
    "value": "[07:09:13.1081, -010.36.38.5952]",
  },
  Object {
    "name": "Image ref coords (deg)",
    "value": "[107.305 deg, -10.6107 deg]",
  },
  Object {
    "name": "Pixel increment",
    "value": "-14\\", 14\\"",
  },
  Object {
    "name": "Celestial frame",
    "value": "FK5, J2000",
  },
  Object {
    "name": "Pixel unit",
    "value": "MJy/sr",
  },
]
`;

exports[`FILEINFO_FITS_MULTIHDU: Testing if info of an FITS image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : spire500_ext.fits FileInfoExtended of '1': assert FILE_INFO_RESPONSE.file_info_extended.header_entries 1`] = `
Array [
  Object {
    "comment": "Standard FITS",
    "entryType": "INT",
    "name": "SIMPLE",
    "numericValue": 1,
    "value": "T",
  },
  Object {
    "entryType": "INT",
    "name": "BITPIX",
    "numericValue": -64,
    "value": "-64",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS",
    "numericValue": 2,
    "value": "2",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS1",
    "numericValue": 830,
    "value": "830",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS2",
    "numericValue": 870,
    "value": "870",
  },
  Object {
    "comment": "PHYSICAL = PIXEL*BSCALE + BZERO",
    "entryType": "FLOAT",
    "name": "BSCALE",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BZERO",
    "value": "0.000000000000E+00",
  },
  Object {
    "name": "BTYPE",
    "value": "Intensity",
  },
  Object {
    "name": "OBJECT",
  },
  Object {
    "comment": "Brightness (pixel) unit",
    "name": "BUNIT",
    "value": "MJy/sr",
  },
  Object {
    "entryType": "FLOAT",
    "name": "EQUINOX",
    "numericValue": 2000,
    "value": "2000",
  },
  Object {
    "name": "RADESYS",
    "value": "FK5",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LONPOLE",
    "numericValue": 180,
    "value": "1.800000000000E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LATPOLE",
    "numericValue": -10.610720896516849,
    "value": "-1.061072089652E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_1",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_1",
    "value": "-0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_2",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "name": "CTYPE1",
    "value": "RA---TAN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL1",
    "numericValue": 107.30461727023817,
    "value": "1.073046172702E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT1",
    "numericValue": -0.003888888888889,
    "value": "-3.888888888889E-03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX1",
    "numericValue": 371,
    "value": "371",
  },
  Object {
    "name": "CUNIT1",
    "value": "deg",
  },
  Object {
    "name": "CTYPE2",
    "value": "DEC--TAN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL2",
    "numericValue": -10.610720896516849,
    "value": "-1.061072089652E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT2",
    "numericValue": 0.003888888888889,
    "value": "3.888888888889E-03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX2",
    "numericValue": 421,
    "value": "421",
  },
  Object {
    "name": "CUNIT2",
    "value": "deg",
  },
  Object {
    "name": "LONGSTRN",
    "value": "OGIP 1.0",
  },
  Object {
    "name": "EXTNAME",
    "value": "image",
  },
  Object {
    "name": "CLASS___",
    "value": "herschel.ia.dataset.ArrayDataset",
  },
  Object {
    "name": "INFO____",
    "value": "Image",
  },
  Object {
    "name": "DATA____",
    "value": "herschel.ia.numeric.Double2d",
  },
  Object {
    "name": "QTTY____",
    "value": "MJy/sr",
  },
  Object {
    "entryType": "INT",
    "name": "META_0",
    "numericValue": 2,
    "value": "2",
  },
  Object {
    "entryType": "INT",
    "name": "META_1",
    "numericValue": 830,
    "value": "830",
  },
  Object {
    "entryType": "INT",
    "name": "META_2",
    "numericValue": 870,
    "value": "870",
  },
  Object {
    "comment": "Time system for HDU",
    "name": "TIMESYS",
    "value": "UTC",
  },
]
`;

exports[`FILEINFO_FITS_MULTIHDU: Testing if info of an FITS image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : spire500_ext.fits FileInfoExtended of '6': assert FILE_INFO_RESPONSE.file_info_extended.computed_entries 1`] = `
Array [
  Object {
    "name": "Name",
    "value": "spire500_ext.fits",
  },
  Object {
    "name": "HDU",
    "value": "6",
  },
  Object {
    "name": "Extension name",
    "value": "error",
  },
  Object {
    "name": "Shape",
    "value": "[830, 870]",
  },
  Object {
    "name": "Coordinate type",
    "value": "Right Ascension, Declination",
  },
  Object {
    "name": "Projection",
    "value": "TAN",
  },
  Object {
    "name": "Image reference pixels",
    "value": "[371, 421]",
  },
  Object {
    "name": "Image reference coords",
    "value": "[07:09:13.1081, -010.36.38.5952]",
  },
  Object {
    "name": "Image ref coords (deg)",
    "value": "[107.305 deg, -10.6107 deg]",
  },
  Object {
    "name": "Pixel increment",
    "value": "-14\\", 14\\"",
  },
  Object {
    "name": "Celestial frame",
    "value": "FK5, J2000",
  },
  Object {
    "name": "Pixel unit",
    "value": "MJy/sr",
  },
]
`;

exports[`FILEINFO_FITS_MULTIHDU: Testing if info of an FITS image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : spire500_ext.fits FileInfoExtended of '6': assert FILE_INFO_RESPONSE.file_info_extended.header_entries 1`] = `
Array [
  Object {
    "comment": "Standard FITS",
    "entryType": "INT",
    "name": "SIMPLE",
    "numericValue": 1,
    "value": "T",
  },
  Object {
    "entryType": "INT",
    "name": "BITPIX",
    "numericValue": -64,
    "value": "-64",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS",
    "numericValue": 2,
    "value": "2",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS1",
    "numericValue": 830,
    "value": "830",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS2",
    "numericValue": 870,
    "value": "870",
  },
  Object {
    "comment": "PHYSICAL = PIXEL*BSCALE + BZERO",
    "entryType": "FLOAT",
    "name": "BSCALE",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BZERO",
    "value": "0.000000000000E+00",
  },
  Object {
    "name": "BTYPE",
    "value": "Intensity",
  },
  Object {
    "name": "OBJECT",
  },
  Object {
    "comment": "Brightness (pixel) unit",
    "name": "BUNIT",
    "value": "MJy/sr",
  },
  Object {
    "entryType": "FLOAT",
    "name": "EQUINOX",
    "numericValue": 2000,
    "value": "2000",
  },
  Object {
    "name": "RADESYS",
    "value": "FK5",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LONPOLE",
    "numericValue": 180,
    "value": "1.800000000000E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LATPOLE",
    "numericValue": -10.610720896516849,
    "value": "-1.061072089652E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_1",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_1",
    "value": "-0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_2",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "name": "CTYPE1",
    "value": "RA---TAN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL1",
    "numericValue": 107.30461727023817,
    "value": "1.073046172702E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT1",
    "numericValue": -0.003888888888889,
    "value": "-3.888888888889E-03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX1",
    "numericValue": 371,
    "value": "371",
  },
  Object {
    "name": "CUNIT1",
    "value": "deg",
  },
  Object {
    "name": "CTYPE2",
    "value": "DEC--TAN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL2",
    "numericValue": -10.610720896516849,
    "value": "-1.061072089652E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT2",
    "numericValue": 0.003888888888889,
    "value": "3.888888888889E-03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX2",
    "numericValue": 421,
    "value": "421",
  },
  Object {
    "name": "CUNIT2",
    "value": "deg",
  },
  Object {
    "name": "LONGSTRN",
    "value": "OGIP 1.0",
  },
  Object {
    "name": "EXTNAME",
    "value": "error",
  },
  Object {
    "name": "CLASS___",
    "value": "herschel.ia.dataset.ArrayDataset",
  },
  Object {
    "name": "INFO____",
    "value": "Statistical error on the pixel values",
  },
  Object {
    "name": "DATA____",
    "value": "herschel.ia.numeric.Double2d",
  },
  Object {
    "name": "QTTY____",
    "value": "MJy/sr",
  },
  Object {
    "entryType": "INT",
    "name": "META_0",
    "numericValue": 2,
    "value": "2",
  },
  Object {
    "entryType": "INT",
    "name": "META_1",
    "numericValue": 830,
    "value": "830",
  },
  Object {
    "entryType": "INT",
    "name": "META_2",
    "numericValue": 870,
    "value": "870",
  },
  Object {
    "comment": "Time system for HDU",
    "name": "TIMESYS",
    "value": "UTC",
  },
]
`;

exports[`FILEINFO_FITS_MULTIHDU: Testing if info of an FITS image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : spire500_ext.fits FileInfoExtended of '7': assert FILE_INFO_RESPONSE.file_info_extended.computed_entries 1`] = `
Array [
  Object {
    "name": "Name",
    "value": "spire500_ext.fits",
  },
  Object {
    "name": "HDU",
    "value": "7",
  },
  Object {
    "name": "Extension name",
    "value": "coverage",
  },
  Object {
    "name": "Shape",
    "value": "[830, 870]",
  },
  Object {
    "name": "Coordinate type",
    "value": "Right Ascension, Declination",
  },
  Object {
    "name": "Projection",
    "value": "TAN",
  },
  Object {
    "name": "Image reference pixels",
    "value": "[371, 421]",
  },
  Object {
    "name": "Image reference coords",
    "value": "[07:09:13.1081, -010.36.38.5952]",
  },
  Object {
    "name": "Image ref coords (deg)",
    "value": "[107.305 deg, -10.6107 deg]",
  },
  Object {
    "name": "Pixel increment",
    "value": "-14\\", 14\\"",
  },
  Object {
    "name": "Celestial frame",
    "value": "FK5, J2000",
  },
  Object {
    "name": "Pixel unit",
    "value": "1",
  },
]
`;

exports[`FILEINFO_FITS_MULTIHDU: Testing if info of an FITS image file is correctly delivered by the backend Go to "set_QA" folder query the info of file : spire500_ext.fits FileInfoExtended of '7': assert FILE_INFO_RESPONSE.file_info_extended.header_entries 1`] = `
Array [
  Object {
    "comment": "Standard FITS",
    "entryType": "INT",
    "name": "SIMPLE",
    "numericValue": 1,
    "value": "T",
  },
  Object {
    "entryType": "INT",
    "name": "BITPIX",
    "numericValue": -64,
    "value": "-64",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS",
    "numericValue": 2,
    "value": "2",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS1",
    "numericValue": 830,
    "value": "830",
  },
  Object {
    "entryType": "INT",
    "name": "NAXIS2",
    "numericValue": 870,
    "value": "870",
  },
  Object {
    "comment": "PHYSICAL = PIXEL*BSCALE + BZERO",
    "entryType": "FLOAT",
    "name": "BSCALE",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "BZERO",
    "value": "0.000000000000E+00",
  },
  Object {
    "name": "BTYPE",
    "value": "Intensity",
  },
  Object {
    "name": "OBJECT",
  },
  Object {
    "comment": "Brightness (pixel) unit",
    "name": "BUNIT",
    "value": "1",
  },
  Object {
    "entryType": "FLOAT",
    "name": "EQUINOX",
    "numericValue": 2000,
    "value": "2000",
  },
  Object {
    "name": "RADESYS",
    "value": "FK5",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LONPOLE",
    "numericValue": 180,
    "value": "1.800000000000E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "LATPOLE",
    "numericValue": -10.610720896516849,
    "value": "-1.061072089652E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_1",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_1",
    "value": "-0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC1_2",
    "value": "0.000000000000E+00",
  },
  Object {
    "entryType": "FLOAT",
    "name": "PC2_2",
    "numericValue": 1,
    "value": "1.000000000000E+00",
  },
  Object {
    "name": "CTYPE1",
    "value": "RA---TAN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL1",
    "numericValue": 107.30461727023817,
    "value": "1.073046172702E+02",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT1",
    "numericValue": -0.003888888888889,
    "value": "-3.888888888889E-03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX1",
    "numericValue": 371,
    "value": "371",
  },
  Object {
    "name": "CUNIT1",
    "value": "deg",
  },
  Object {
    "name": "CTYPE2",
    "value": "DEC--TAN",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRVAL2",
    "numericValue": -10.610720896516849,
    "value": "-1.061072089652E+01",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CDELT2",
    "numericValue": 0.003888888888889,
    "value": "3.888888888889E-03",
  },
  Object {
    "entryType": "FLOAT",
    "name": "CRPIX2",
    "numericValue": 421,
    "value": "421",
  },
  Object {
    "name": "CUNIT2",
    "value": "deg",
  },
  Object {
    "name": "LONGSTRN",
    "value": "OGIP 1.0",
  },
  Object {
    "name": "EXTNAME",
    "value": "coverage",
  },
  Object {
    "name": "CLASS___",
    "value": "herschel.ia.dataset.ArrayDataset",
  },
  Object {
    "name": "INFO____",
    "value": "Coverage",
  },
  Object {
    "name": "DATA____",
    "value": "herschel.ia.numeric.Double2d",
  },
  Object {
    "name": "QTTY____",
    "value": "1",
  },
  Object {
    "entryType": "INT",
    "name": "META_0",
    "numericValue": 2,
    "value": "2",
  },
  Object {
    "entryType": "INT",
    "name": "META_1",
    "numericValue": 830,
    "value": "830",
  },
  Object {
    "entryType": "INT",
    "name": "META_2",
    "numericValue": 870,
    "value": "870",
  },
  Object {
    "comment": "Time system for HDU",
    "name": "TIMESYS",
    "value": "UTC",
  },
]
`;
